Spigot-Server-Patches/0619-Added-SoundBuilder.patch

```diff
@@ -0,0 +1,438 @@
++    @Override
++    public Sound sound() {
++        return sound;
++    }
++
++    /**
++     * Changes what sound will be sent
++     *
++     * @param sound The sound
++     * @return a reference to this object.
++     */
++    @Override
++    public PaperSoundBuilder sound(Sound sound) {
++        this.sound = sound;
++        return this;
++    }
++
++    /**
++     * @return The category under which the sound will be played
++     */
++    @Override
++    public SoundCategory category() {
++        return category;
++    }
++
++    /**
++     * Changes under which category the sound will be played
++     */
++    @Override
++    public PaperSoundBuilder category(SoundCategory category) {
++        this.category = category;
++        return this;
++    }
++
++    /**
++     * @return The volume level of the sound played
++     */
++    @Override
++    public Float volume() {
++        return volume;
++    }
++
++    /**
++     * Changes the volume level of the sound played
++     *
++     * @param volume The volume
++     * @return a reference to this object.
++     */
++    @Override
++    public PaperSoundBuilder volume(Float volume) {
++        this.volume = volume;
++        return this;
++    }
++
++    /**
++     * @return The pitch of the sound played
++     */
++    @Override
++    public Float pitch() {
++        return pitch;
++    }
++
++    /**
++     * Changes the pitch of the sound played
++     *
++     * @param pitch The pitch
++     * @return a reference to this object.
++     */
++    @Override
++    public PaperSoundBuilder pitch(Float pitch) {
++        this.pitch = pitch;
++        return this;
++    }
++
++    /**
++     * @return List of players who will receive the sound, or null for all in world
++     */
++    @Override
++    public List<Player> receivers() {
++        return Lists.newLinkedList(receivers);
++    }
++
++    /**
++     * Example use:
++     *
++     * builder.receivers(16)
++     *
++     * @return If this sound is going to be sent to someone
++     */
++    @Override
++    public boolean hasReceivers() {
++        return (receivers == null && !location.getWorld().getPlayers().isEmpty()) || (
++            receivers != null && !receivers.isEmpty());
++    }
++
++    /**
++     * Sends this sound to all players in the world. This is rather silly and you should likely not
++     * be doing this.
++     *
++     * Just be a logical person and use receivers by radius or collection.
++     *
++     * @return a reference to this object.
++     */
++    @Override
++    public PaperSoundBuilder allPlayers() {
++        this.receivers = null;
++        return this;
++    }
++
++    /**
++     * @param receivers List of players to receive this sound, or null for all players in the
++     * world
++     * @return a reference to this object.
++     * @deprecated The use of list's is not appreciated
++     */
++    @Override
++    @Deprecated
++    public PaperSoundBuilder receivers(List<Player> receivers) {
```

Comment:
The soundbuilder api is a new one. Just drop the method no need to introduce a method that is deprecated by default.